### PR TITLE
Make ILSpan internal

### PR DESCRIPTION
**Pull request by tmat**
https://github.com/dotnet/roslyn/issues/9052

### CODE REVIEW

**Comment by tmat** [comment]
Fixes https://github.com/dotnet/roslyn/issues/9052

File: src/ExpressionEvaluator/Core/Source/ExpressionCompiler/ILSpan.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 namespace Microsoft.CodeAnalysis.ExpressionEvaluator
 {
     // TODO: use TextSpan?
-    public struct ILSpan : IEquatable<ILSpan>
+    internal struct ILSpan : IEquatable<ILSpan>
```
Comment (jaredpar) [inline src/ExpressionEvaluator/Core/Source/ExpressionCompiler/ILSpan.cs]: This should cause a break in the pubilc API checker ...

Comment (tmat) [inline src/ExpressionEvaluator/Core/Source/ExpressionCompiler/ILSpan.cs]: Looks like we don't run it for ExpressionCompiler. That's likely how this got unnoticed in the first place. Will follow up with adding the public api files.

**Comment by jaredpar** [comment]
:+1:

**Comment by MattGertz** [comment]
LGTM, approved.

**Comment by tmat** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/4276/
// Retest reason: https://github.com/dotnet/roslyn/issues/9166

**Pull request merged**